### ISSUE TITLE
Editable Dropdown broken in 13.4.0

### ISSUE THREAD

**Issue by Cito** [primefaces/primeng#11522]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**

https://stackblitz.com/edit/primeng-13-4-0-dropdown-issue

**Current behavior**

When using a Dropdown component with `editable="true"` in PrimeNG 13.4.0, and a value is entered in the editable input field via the keyboard that does not exist as one of the predefined options, the value is output as null.

**Expected behavior**

When `editable="true"`, it should be possible to enter any custom value instead of the predefined options using the editable input field.

**Minimal reproduction of the problem with instructions**

See StackBlitz demo above.
- When you set `"primeng": "13.3.3",` in package.json, it works as expected: The entered value is reflected above the input after "Selected value:".
- When you set  `"primeng": "13.4.0",` in package.json, custom values can be entered, but there is no output for these, and "Selected value:" shows only "None" for these.

**What is the motivation / use case for changing the behavior?**

Restoring the proper and documented behavior of 13.3.3.

**Suspected cause / suggested fix**:

I believe this issue is caused by ed3b3088a67f8f1e59f365b240d4df1b78e2ebaa. Since it has only a non-descriptive commit message "update dropdown.ts", I do not understand what the intention of this change was. In any case, in the `else if` expression that was inserted in this commit, a `&& !this.editable` must be probably appended, or the whole `if - else if` chain should be wrapped into an `if (!this.editable)`.

* **Angular version:** 13.3.8.

* **PrimeNG version:** 13.4.0.

* **Browser:** all
 
* **Language:** all

**Comment by agencia-estado** [comment]
I have the same issue. For now, I will use version 13.3.8

**Comment by cetincakiroglu** [comment]
Hi, 

Thanks for your report, this issue is fixed with the current release (13.4.1).

Regards
